pkg/sql/executor_ir.go

```diff
@@ -112,33 +114,57 @@ func submitWorkflow(wr *PipeWriter, sqlProgram string, db *DB, modelDir string,
 		return err
 	}
 
-	connStr := fmt.Sprintf("%s://%s", db.driverName, db.dataSourceName)
+	spIRs, err := genSQLProgramIR(
+		sqls, db, modelDir,
+		false, /*disable feature derivation*/
+		false /*disable load trainIR from saved model and verify with trainIR*/)
 
-	_ = sqls
-	_ = connStr
-	/*
-		_, err = programToIR(sqls, connStr, modelDir, true, false)
-		if err != nil {
-			return err
-		}
-	*/
+	if err != nil {
+		return err
+	}
 
-	// TODO(yancey1989):
 	// 1. call codegen_couler.go to genearte Couler program.
+	program, err := couler.Run(spIRs)
+	if err != nil {
+		return fmt.Errorf("Generate Couler program error: %v", err)
+	}
+
+	coulerFile, err := ioutil.TempFile("/tmp", "sqlflow-couler*.py")
+	if err != nil {
+		return fmt.Errorf("")
+	}
+	coulerFile.Write([]byte(program))
+	defer coulerFile.Close()
+
 	// 2. compile Couler program into Argo YAML.
+	argoYaml, err := ioutil.TempFile("/tmp", "sqlflow-argo*.yaml")
+	cmd := exec.Command("couler", "run", "--mode", "argo", "--file", coulerFile.Name(), ">", argoYaml.Name())
+	if _, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("generate Argo workflow yaml error: %v", err)
+	}
+
 	// 3. submit Argo YAML and fetch the workflow ID.
+	cmd = exec.Command("kubectl", "create", "-f", argoYaml.Name())
```

Comment:
can use k8s API here?

Reply:
Sure, will import k8s package when implement `Fetch` gRPC interface, and I will rewrite here to use k8s API.